src/LinkDotNet.Blog.Web/RegistrationExtensions/SqlRegistrationExtensions.cs

```diff
@@ -61,4 +61,21 @@ public static void UseMySqlAsStorageProvider(this IServiceCollection services)
         });
         services.AddScoped(typeof(IRepository<>), typeof(Repository<>));
     }
+
+    public static void UsePostgresAsStorageProvider(this IServiceCollection services)
+    {
+        services.AssertNotAlreadyRegistered(typeof(IRepository<>));
+
+        services.AddPooledDbContextFactory<BlogDbContext>(
+            (s, builder) =>
+            {
+                var configuration = s.GetRequiredService<AppConfiguration>();
+                var connectionString = configuration.ConnectionString;
+                builder.UseNpgsql(connectionString)
+#if DEBUG
+                .EnableDetailedErrors()
+#endif
+                ;
+            });
```

Comment:
```csharp
services.AddScoped(typeof(IRepository<>), typeof(Repository<>));
```

This line is missing - that is why you get the exception on startup.

Reply:
Thank you. My bad.

Let me use this comment to enrich my knowledge... 

Is it a generic injection? I had never seen it before using empty tags.

```csharp
services.AddScoped(typeof(IRepository<>), typeof(Repository<>));
```

Reply:
Yes - it is called open-type registration